core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/CollectInformation.scala

```diff
@@ -81,11 +81,54 @@ class CollectInformation(apps: Seq[ApplicationInfo]) extends Logging {
 
   // get read data schema information
   def getDataSourceInfo: Seq[DataSourceProfileResult] = {
-    val filtered = apps.filter(_.dataSourceInfo.size > 0)
-    val allRows = filtered.flatMap { app =>
-      app.dataSourceInfo.map { ds =>
-        DataSourceProfileResult(app.index, ds.sqlID, ds.format, ds.location,
-          ds.pushedFilters, ds.schema)
+    val dataSourceApps = apps.filter(_.dataSourceInfo.size > 0)
+    val sqlAccums = CollectInformation.generateSQLAccums(dataSourceApps)
+    // Metrics to capture from event log to the result
+    val buffer_time: String = "buffer time"
+    val scan_time = "scan time"
+    val data_size = "data size"
+    val decode_time = "GPU decode time"
+
+    // This is to save the metrics which will be extracted while creating the result.
+    case class IoMetrics(
+        buffer_time: String,
+        scan_time: String,
+        data_size: String,
+        decode_time: String)
+
+    def getIoMetrics(metricsName: String, metricsValue: String): IoMetrics = {
+      metricsName match {
+        case value if (value == buffer_time) => IoMetrics(metricsValue, "", "", "")
+        case value if (value == scan_time) => IoMetrics("", metricsValue, "", "")
+        case value if (value == data_size) => IoMetrics("", "", metricsValue, "")
+        case value if (value == decode_time) => IoMetrics("", "", "", metricsValue)
+      }
+    }
+
+    val allRows = dataSourceApps.flatMap { app =>
+      val appSqlAccums = sqlAccums.filter(x => x.appIndex == app.index)
+      // Number of data source reads in an application
+      app.readCount = appSqlAccums.filter(x => x.name.contains(scan_time)).size
+
+      //Filter appSqlAccums to get only required metrics
+      val dataSourceMetrics = appSqlAccums.filter(x => x.name.contains(buffer_time)
+          || x.name.contains(scan_time) || x.name.contains(decode_time)
+          || x.name.equals(data_size))
+
+      app.dataSourceInfo.flatMap { ds =>
+        val sqlIdtoDs = dataSourceMetrics.filter(x => x.sqlID == ds.sqlID)
+        if (!sqlIdtoDs.isEmpty) {
+          sqlIdtoDs.map { x =>
+            val ioMetrics = getIoMetrics(x.name, x.max_value.toString)
```

Comment:
I'm quite confused by this, we get the results of a single metric but then create a DataSourceProfileResult with empty strings for some of them?  DataSourceProfileResults to me sounds like the entire datasource information.  Upon reading this I don't see how this is actually working unless its coalesced later which doesn't make sense to me

Reply:
In order to produce the result in csv format i.e  separate columns for data_size, decode_time, buffer time and so on, only one column will have value and rest of them will be blank. In the above code, each row of DataSourceProfileResults will contain one of the values from sqlIdToDs. Posted the output from one of the runs for your reference. Please let me know if the output needs to be presented in a different way.

```
appIndex,sqlID,readCount,format,location,pushedFilters,schema,accumId,buffer_time,scan_time,data_size,decode_time
1,0,-1,Text,InMemoryFileIndex[file:/home/user1/workspace/spark-rapids-another/integration_tests/src/test/resources/people.csv],[],value:string,NA,NA,NA,NA,NA
1,1,4,CSV(GPU),file:/home/user1/workspace/spark-rapids-another/integration_tests/src/test/re...,],_c0:string;_c1:string;_c2:string,60,"",26,"",""
1,1,4,CSV(GPU),file:/home/user1/workspace/spark-rapids-another/integration_tests/src/test/re...,],_c0:string;_c1:string;_c2:string,57,"","","",8465370
1,1,4,CSV(GPU),file:/home/user1/workspace/spark-rapids-another/integration_tests/src/test/re...,],_c0:string;_c1:string;_c2:string,58,1180511,"","",""
1,2,4,JSON(GPU),file:/home/user1/workspace/spark-rapids-another/integration_tests/target/test...,],number:double,125,"",22,"",""
1,2,4,JSON(GPU),file:/home/user1/workspace/spark-rapids-another/integration_tests/target/test...,],number:double,122,"","","",2703739
1,2,4,JSON(GPU),file:/home/user1/workspace/spark-rapids-another/integration_tests/target/test...,],number:double,123,305515,"","",""

```

Reply:
I'm not understanding why we want these on separate lines, lets discuss further

Reply:
Updated the output format to be one row per sqlId